File: PROYECTOPOW-SEBASTIAN PEJENDINO ACEVEDO/src/com/nombrecolegio/gestor/model/Usuario.java
package com.nombrecolegio.gestor.model;

public class Usuario {
    private String nombre;
    private String email;
    private String contraseña;
    private String rol; // Puede ser "ADMIN", "PROFESOR", "ESTUDIANTE"

    public Usuario(String nombre, String email, String contraseña, String rol) {
        this.nombre = nombre;
        this.email = email;
        this.contraseña = contraseña;
        this.rol = rol;
    }

    // Getters y Setters
    public String getNombre() {
        return nombre;
    }

    public void setNombre(String nombre) {
        this.nombre = nombre;
    }

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public String getContraseña() {
        return contraseña;
    }

    public void setContraseña(String contraseña) {
        this.contraseña = contraseña;
    }

    public String getRol() {
        return rol;
    }

    public void setRol(String rol) {
        this.rol = rol;
    }
}

File: POW/Models/Usuario.cs
namespace Pow.Models
{
    public class Usuario
    {
        public int Id { get; set; }
        public string Email { get; set; }
        public string PasswordHash { get; set; }
        public string Nombre { get; set; }
        public int RolId { get; set; }  // Esta propiedad debe ser establecida

        // Propiedades de navegación
        public Rol Rol { get; set; }
        public Profesor Profesor { get; set; }
        public Estudiante Estudiante { get; set; }
    }
}
